### PR TITLE
feat(self-managed): document multi-tenancy enabled flag

### PR DESCRIPTION
**Pull request by nikku**
## Description

This adds the available `CAMUNDA_OPERATE_MULTITENANCY_ENABLED` env flag to the documentation.

Related to [this discussion](https://camunda.slack.com/archives/C9B5270DA/p1702288473663209).

Changes mirror naming in [other product components](https://docs.camunda.io/optimize/self-managed/optimize-deployment/configuration/multi-tenancy/#configuration). 

> :information_source: Note that you seem to swap `default value` and description (compared to other components). I did not clean that up.

## When should this change go live?

- [ ] This change is not yet live and should not be merged until {ADD_DATE} (apply `hold` label or convert to draft PR)?
- [x] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [x] This functionality is already available but undocumented.
- [ ] This is a bug fix or security concern.

## PR Checklist

- [x] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [x] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [ ] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Comment by akeller** [comment]
This seems very similar to what I worked through with Tasklist on #2997. 

I ask that the Operate team review this PR in the context of the entire page. This is the only table that includes an environment var (in #2997, we moved the environment var into the description since the top of the page says the config is YAML This matches how Zeebe does it.).

We should aim for consistency on the page and across the components.

File: docs/self-managed/operate-deployment/operate-configuration.md [hunk 1]
```diff
@@ -62,9 +62,9 @@ the feature can be found in [the multi-tenancy documentation](../concepts/multi-
 
 The following configuration is required to enable multi-tenancy in Operate:
 
-| Name                                 | Description                                         | Default value |
-| ------------------------------------ | --------------------------------------------------- | ------------- |
-| camunda.operate.multiTenancy.enabled | Activates the multi-tenancy feature within Operate. | false         |
+| YAML path                            | Environment variable                 | Description                                         | Default value |
+| ------------------------------------ | ------------------------------------ | --------------------------------------------------- | ------------- |
+| camunda.operate.multiTenancy.enabled | CAMUNDA_OPERATE_MULTITENANCY_ENABLED | Activates the multi-tenancy feature within Operate. | false         |
```
Comment (koevskinikola) [inline docs/self-managed/operate-deployment/operate-configuration.md:67]: :wrench: If we're keeping the table format aligned across components, the env var name should be included in the description. Similar to Tasklist, we already mention at the beginning of this page that Operate is a Spring Boot application, and the standard configuration mechanisms of a Spring Boot application can be used.

File: versioned_docs/version-8.3/self-managed/operate-deployment/operate-configuration.md [hunk 2]
```diff
@@ -62,9 +62,9 @@ the feature can be found in [the multi-tenancy documentation](../concepts/multi-
 
 The following configuration is required to enable multi-tenancy in Operate:
 
-| Name                                 | Description                                         | Default value |
-| ------------------------------------ | --------------------------------------------------- | ------------- |
-| camunda.operate.multiTenancy.enabled | Activates the multi-tenancy feature within Operate. | false         |
+| YAML path                            | Environment variable                 | Description                                         | Default value |
+| ------------------------------------ | ------------------------------------ | --------------------------------------------------- | ------------- |
+| camunda.operate.multiTenancy.enabled | CAMUNDA_OPERATE_MULTITENANCY_ENABLED | Activates the multi-tenancy feature within Operate. | false         |
```
Comment (koevskinikola) [inline versioned_docs/version-8.3/self-managed/operate-deployment/operate-configuration.md:67]: :wrench: Same as above.

**Review by koevskinikola — CHANGES_REQUESTED** [review]
Just a small suggestion about the formatting of the tables for consistency. 

:x: We don't really define env vars in the rest of the tables on this page. So introducing a separate column will be inconsistent.

File: docs/self-managed/operate-deployment/operate-configuration.md (hunk 1 above)
Comment (nikku) [reply docs/self-managed/operate-deployment/operate-configuration.md:67]: > [...] we already mention at the beginning of this page that Operate is a Spring Boot application, and the standard configuration mechanisms of a Spring Boot application can be used

This implies that people _know_ what spring boot is, and how it works. Let's not force folks to learn this as they try to configure Camunda. My two cents.

**Comment by christinaausley** [comment]
@nikku Can I assist with this? Would you like to share this in #ask-operate for someone to pick this up?

I know folks are out on FTO, so no worries if we just haven't circled back on this yet. Just checking in.

**Comment by nikku** [comment]
@christinaausley Yes, please assist me.

We can either merge this PR (with my changes), consider small adjustments (I'm OK to incorporate them) or rework the whole page (forward to Operate folks). Your call.

**Comment by christinaausley** [comment]
@nikku I'm fine with merging in these smaller adjustments and iterating, whether that be via additional small adjustments, or a complete rework by the Operate team. CC @ralfpuchert

**Comment by akeller** [comment]
> This seems very similar to what I worked through with Tasklist on #2997.
> 
> I ask that the Operate team review this PR in the context of the entire page. This is the only table that includes an environment var (in #2997, we moved the environment var into the description since the top of the page says the config is YAML This matches how Zeebe does it.).
> 
> We should aim for consistency on the page and across the components.

@johanwelgemoed can you chime in here? Adding the env var is important, but @nikku makes a good point about not requiring the user to learn or know Spring Boot. This is likely a bigger topic for all our Spring Boot applications. 

If this is still open by European time zones EOD tomorrow, I will rework this PR to match the way I handled this with Tasklist, keeping the page consistent but adding the env var to the description.

**Comment by sdorokhova** [comment]
@akeller Johan asked me to look into this. I agree with Christina that we can merge this PR (or feel free to rework the way you prefer) and then iteratively improve the other parts of the docs. I created [this issue](https://github.com/camunda/camunda-docs/issues/3165) to document all parameters in both formats. Also feel fee to adjust the issue if you have your opinion on that.

**Review by akeller — APPROVED** [review]
Let's merge as-is. Thanks everyone!

**Pull request merged**